gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/strategy/decoration/SubgraphStrategy.java

```diff
@@ -105,21 +105,11 @@ private SubgraphStrategy(final Traversal<Vertex, ?> vertexCriterion, final Trave
         this.vertexPropertyCriterion = null == vertexPropertyCriterion ? null : vertexPropertyCriterion.asAdmin().clone();
 
         if (null != this.vertexCriterion)
-            this.addLabelMarker(this.vertexCriterion);
+            TraversalHelper.applyTraversalRecursively(t -> t.getStartStep().addLabel(MARKER), this.vertexCriterion);
```

Comment:
`vertexCriterion` can be `null`. In that case `TraversalHelper.applyTraversalRecursively` and/or the consumer would throw a NPE.